src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo);
+        final int index0 = 0;
+        final int index1 = 1;
+        final int index2 = 2;
+        final int index3 = 3;
+        for (String name : names) {
+            int income = 0;
+            for (String dataForSplit : data) {
+                String[] dataArray = dataForSplit.split(" ");
+                if (name.equals(dataArray[index1]) && parseDateToInt(dataArray[index0])
```

Comment:
Check the period between dates using LocalDate methods such as `isAfter()`, `isBefore()`, and etc.